src/main/java/com/google/api/pathtemplate/PathTemplate.java

```diff
@@ -132,6 +132,9 @@
   // A regexp to match a custom verb at the end of a path.
   private static final Pattern CUSTOM_VERB_PATTERN = Pattern.compile(":([^/*}{=]+)$");
 
+  // A regex to match a hostname with or without protocol.
+  private static final Pattern HOSTNAME_PATTERN = Pattern.compile("^(\\w+:)?//");
```

Comment:
Now that `HOSTNAME_PATTERN` also matches with a protocol, can you update the comment for `HOSTNAME_VAR` to explain that that might also contain a protocol string?